tensorflow_encrypted/protocol/pond.py

```diff
@@ -585,6 +586,33 @@ def stack(self, xs: List['PondTensor'], axis: int = 0):
 
         return xs_stack
 
+    def concat(self, xc: List['PondTensor'], axis):
+
+        node_key = ('concat', tuple(xc))
+        xc_concat = nodes.get(node_key, None)
+
+        if xc_concat is not None:
+            return xc_concat
+
+        if all([isinstance(x, PondPublicTensor) for x in xc]):
+            xc_concat = _concat_shared(self, xc, axis=axis)
+
+        elif all([isinstance(x, PondPrivateTensor) for x in xc]):
+            xc_concat = _concat_shared(self, xc, axis=axis)
+
+        elif all([isinstance(x, PondMaskedTensor) for x in xc]):
+            xc_concat = _concat_masked(self, xc, axis=axis)
+
+            unmasked = [x.unmasked for x in xc_concat]
```

Comment:
no need for these two lines anymore since `_concat_masked` calls back into `prot.stack` for `x.unmasked`